garminexport/garminbackup.py

```diff
@@ -0,0 +1,129 @@
+#! /usr/bin/env python
+"""Performs (incremental) backups of activities for a given Garmin Connect
+account.
+The activities are stored in a local directory on the user's computer.
+The backups are incremental, meaning that only activities that aren't already
+stored in the backup directory will be downloaded.
+"""
+import argparse
+from datetime import timedelta
+import getpass
+from garminexport.garminclient import GarminClient
+import garminexport.backup
+from garminexport.backup import export_formats
+from garminexport.retryer import (
+    Retryer, ExponentialBackoffDelayStrategy, MaxRetriesStopStrategy)
+import logging
+import os
+import re
+import sys
+import traceback
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)-15s [%(levelname)s] %(message)s")
+log = logging.getLogger(__name__)
+
+LOG_LEVELS = {
+    "DEBUG": logging.DEBUG,
+    "INFO": logging.INFO,
+    "WARNING": logging.WARNING,
+    "ERROR": logging.ERROR
+}
+"""Command-line (string-based) log-level mapping to logging module levels."""
+
+DEFAULT_MAX_RETRIES = 7
+"""The default maximum number of retries to make when fetching a single activity."""
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser(
+        description=(
+            "Performs incremental backups of activities for a "
+            "given Garmin Connect account. Only activities that "
+            "aren't already stored in the backup directory will "
+            "be downloaded."))
+    # positional args
+    parser.add_argument(
+        "username", metavar="<username>", type=str, help="Account user name.")
+    # optional args
+    parser.add_argument(
+        "--password", type=str, help="Account password.")
+    parser.add_argument(
+        "--backup-dir", metavar="DIR", type=str,
+        help=("Destination directory for downloaded activities. Default: "
+              "./activities/"), default=os.path.join(".", "activities"))
+    parser.add_argument(
+        "--log-level", metavar="LEVEL", type=str,
+        help=("Desired log output level (DEBUG, INFO, WARNING, ERROR). "
+              "Default: INFO."), default="INFO")
+    parser.add_argument(
+        "-f", "--format", choices=export_formats,
+        default=None, action='append',
+        help=("Desired output formats ("+', '.join(export_formats)+"). "
+              "Default: ALL."))
+    parser.add_argument(
+        "-E", "--ignore-errors", action='store_true',
+        help="Ignore errors and keep going. Default: FALSE")
+    parser.add_argument(
+        "--max-retries", metavar="NUM", default=DEFAULT_MAX_RETRIES,
+        type=int, help="The maximum number of retries to make on failed attempts to fetch an activity. Exponential backoff will be used, meaning that the delay between successive attempts will double with every retry, starting at one second. DEFAULT: %d" % DEFAULT_MAX_RETRIES)
+
+    return parser.parse_args()
+
+
+def garminbackup(args=None):
+
```

Comment:
This PR aims to make `garminbackup` a publically exposed API. IMHO, a "anonymous" dict doesn't cut it for an API. We need to be explicit about the arguments. Add one key-value argument for every parameter that can be passed as an arg and document the function API.

Reply:
Also, describe what the function does and where/how it stores downloaded content.

Reply:
Also, I wonder if `incremental_backup()` would be a better name.

Reply:
changed the name of the function. parametrized it with the same arguments as requested by arg parser